### PR TITLE
ActorSystems in REPL

### PR DESCRIPTION
**Pull request by lustefaniak**

### CODE REVIEW

File: demo/src/main/scala/Main.scala [hunk 1]
```diff
@@ -28,4 +30,69 @@ object Main extends App {
   for (i <- 0 to 1000) {
     lazyActor1.tell("doit", lazyActor2)
   }
+
+  class TreeActor extends Actor {
```
Comment (JJag) [inline demo/src/main/scala/Main.scala]: I think it's a good idea to move it to seperate file

File: monitoring/src/main/scala/akkaviz/events/ActorSystems.scala [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package akkaviz.events
+
+import akka.actor.ActorSystem
+
+import scala.ref.WeakReference
+
+object ActorSystems {
+
+  private val _systems = scala.collection.mutable.Map[String, WeakReference[ActorSystem]]()
```
Comment (lustefaniak) [inline monitoring/src/main/scala/akkaviz/events/ActorSystems.scala]: Codacy Issue found: [Field name does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2106955461/issues/source?bid=3121282&fileBranchId=3144103#l9)

Comment (lustefaniak) [inline monitoring/src/main/scala/akkaviz/events/ActorSystems.scala]: Codacy Issue found: [Field name '_systems' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2106955461/issues/source?bid=3121282&fileBranchId=3144103#l9)

File: monitoring/src/main/scala/akkaviz/events/ActorSystems.scala [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  def systems: scala.collection.immutable.Map[String, ActorSystem] = _systems.flatMap {
+    case (name, ref) => ref.get.map {
```
Comment (lustefaniak) [inline monitoring/src/main/scala/akkaviz/events/ActorSystems.scala]: Codacy Issue found: [Usage of get on optional type.](https://www.codacy.com/app/lustefaniak/akka-viz/file/2106955461/issues/source?bid=3121282&fileBranchId=3144103#l12)

File: demo/src/main/scala/Main.scala (hunk 1 above)
Comment (lustefaniak) [inline demo/src/main/scala/Main.scala]: Done

File: monitoring/src/main/scala/akkaviz/events/ActorSystems.scala (hunk 3 above)
Comment (lustefaniak) [inline monitoring/src/main/scala/akkaviz/events/ActorSystems.scala]: Stupid Codacy :-/

File: monitoring/src/main/scala/akkaviz/events/ActorSystems.scala [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import akka.actor.ActorSystem
+
+import scala.ref.WeakReference
+
+object ActorSystems {
+
+  private val systemReferences = scala.collection.mutable.Map[String, WeakReference[ActorSystem]]()
+
+  def systems: scala.collection.immutable.Map[String, ActorSystem] = systemReferences.flatMap {
+    case (name, ref) => ref.get.map {
```
Comment (lustefaniak) [inline monitoring/src/main/scala/akkaviz/events/ActorSystems.scala]: Codacy Issue found: [Usage of get on optional type.](https://www.codacy.com/app/lustefaniak/akka-viz/file/2110675929/issues/source?bid=3121282&fileBranchId=3144103#l12)

**Pull request merged**